org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/backend/filebased/ConfigurationBasedNamespaceManager.java

```diff
@@ -36,64 +47,149 @@ public ConfigurationBasedNamespaceManager(Configuration configuration) {
 
         // example namespaces opened for users to create new types
         this.configuration.setProperty(Namespaces.EXAMPLE_NAMESPACE_URI, "ex");
-        this.configuration.setProperty(Namespaces.URI_OPENTOSCA_NODETYPE, "otnt");
     }
 
     @Override
     public String getPrefix(String namespace) {
-        Objects.requireNonNull(namespace);
+        if (namespace == null) {
+            namespace = "";
+        }
+
+        // configuration stores the permanent mapping
+        // this has precedence
         String prefix = configuration.getString(namespace);
-        if (prefix == null) {
-            prefix = this.generatePrefix(namespace);
-            this.configuration.setProperty(namespace, prefix);
+        if (prefix == null || prefix.isEmpty()) {
+            // in case no permanent mapping is found - or the prefix is invalid, check the in-memory ones
+            prefix = this.namespaceToPrefixMap.get(namespace);
+            if (prefix == null) {
+                prefix = this.generatePrefix(namespace);
+                this.namespaceToPrefixMap.put(namespace, prefix);
+            }
         }
         return prefix;
     }
 
     @Override
-    public boolean hasPrefix(String namespace) {
+    public boolean hasPermanentPrefix(String namespace) {
         return this.configuration.containsKey(namespace);
     }
 
     @Override
-    public void remove(String namespace) {
+    public void removePermanentPrefix(String namespace) {
         this.configuration.clearProperty(namespace);
+        // ensure that in-memory mapping also does not have the key any more
+        this.namespaceToPrefixMap.remove(namespace);
     }
 
     @Override
-    public void setPrefix(String namespace, String prefix) {
-        if (!this.getAllPrefixes().contains(prefix)) {
+    public void setPermanentPrefix(String namespace, String prefix) {
+        if (Objects.isNull(namespace) || Objects.isNull(prefix)) {
+            return;
+        }
+        if (namespace.isEmpty() || prefix.isEmpty()) {
+            return;
+        }
+        if (!this.getAllPermanentPrefixes().contains(prefix)) {
             this.configuration.setProperty(namespace, prefix);
+            // ensure that in-memory mapping also does not have the key any more
+            this.namespaceToPrefixMap.remove(namespace);
+        }
+    }
+
+    /**
+     * Generates a string indicating the kind of definitions children maintained by the namespace
+     *
+     * @return empty string if nothing could be matched
+     */
+    private String generateKindString(String namespace) {
```

Comment:
I would rename it to something like `getDefinitionsTypeAbbreveation()`.
`generateKindString()` is too generic  IMHO.